components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultLogoutRequestHandler.java

```diff
@@ -403,13 +403,16 @@ protected void sendResponse(HttpServletRequest request, HttpServletResponse resp
         // attributes
         request.setAttribute(FrameworkConstants.ResponseParams.LOGGED_OUT, isLoggedOut);
 
-
-        if (isLoggedOut && !isValidCallerPath(context)) {
-            if (log.isDebugEnabled()) {
-                log.debug("The commonAuthCallerPath param specified in the request does not satisfy the logout return" +
-                        " url specified. Therefore directing to the default logout return url.");
+        if (Boolean.valueOf(IdentityUtil.getProperty(ENABLE_VALIDATING_LOGOUT_RETURN_URL_CONFIG))) {
+            if (isLoggedOut && !isValidCallerPath(context)) {
+                if (log.isDebugEnabled()) {
```

Comment:
Might not need if check.

Reply:
Based on [1] and [2], it is more efficient to use an if condition to check if debug has been enabled when the `+` operator has been used for String concatenation as in this case.

[1] https://dzone.com/articles/measuring-string-concatenation
[2] https://stackoverflow.com/questions/6504407/is-there-a-need-to-do-a-iflog-isdebugenabled-check

Reply:
Yes. But I doubt we need to see this as a concatenation. This is one string right?

Reply:
Yes, since we are not using any variables/parameters, this doesn't involve the inefficient type of concatenation and therefore, the if check is not necessary. Will fix this.

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/5789.